webclient/app/src/app/features/trafficIncident/IncidentOverviewMap.jsx

```diff
@@ -73,22 +104,66 @@ function IncidentOverviewMap() {
             }
         }),
         // Add layer with Incident Icons
-        new IconLayer({
-            id: 'icon-layer',
-            data: trafficIncidents,
+        new ScatterplotLayer({
+            id: 'scatterplot-layer',
+            data: Object.entries(groupedIncidents),     // Using Object.entries to convert object to array of [key, value] pairs
+
+            pickable: true,     // Enable hover interactions with the icons
+            opacity: 0.8,       // 80% opacity for icons
+            stroked: true,      // Show border around icons
+            filled: true,       // Fill icons with color
+
+            // Size settings for markers
+            radiusScale: 15,
+            radiusMinPixels: 5,
+            radiusMaxPixels: 100,
+            lineWidthMinPixels: 1,
+
+            // Function to determine icon position
+            // d[1][0] contains the first incident in the group (all have same coordinates)
+            getPosition: d => [
+                d[1][0].cameraLongitude,
+                d[1][0].cameraLatitude
+            ],
+            getRadius: d => Math.sqrt(d[1].length) * 5,
+            getFillColor: d => getIconColor(d[1].length),
+            getLineColor: [0, 0, 0, 255],
+            onHover: info => {
+                if (info.object) {
+                    setHoveredIncidents(info.object[1]);
+                }
+            }
+        }),
+        new TextLayer({
+            id: 'text-layer',
+            data: Object.entries(groupedIncidents),
```

Comment:
Why you are using Object.entries here?

Reply:
```jsx
const groupedIncidents = trafficIncidents.reduce((acc, incident) => {
        const key = `${incident.cameraLatitude}-${incident.cameraLongitude}`;   // Create a unique key using the camera coordinates. For example: "39.78-86.15"
        if (!acc[key]) {    // If this is the first incident at these coordinates, initialize an empty array for this location
            acc[key] = []; 
        }
        acc[key].push(incident);    
        return acc;
    }, {});
```
This code snippet groups incidents by their location using the reduce method on the trafficIncidents array.
The purpose of grouping is to combine several incidents that occurred in the same place (with the same coordinates).

Example of what groupedIncidents looks like:
```jsx
groupedIncidents = {
        "39.78-86.15": [        // Key: coordinate string
            {                    // Value: array of incidents at this location
                id: 1,
                cameraLatitude: 39.78,
                cameraLongitude: -86.15,
                state: "NEW",
                description: "Traffic jam"
            },
            {
                id: 2,
                cameraLatitude: 39.78,
                cameraLongitude: -86.15,
                state: "OPEN",
                description: "Accident"
            }
        ],
        "39.80-86.13": [
            {
                id: 3,
                cameraLatitude: 39.80,
                cameraLongitude: -86.13,
                state: "NEW",
                description: "Road work"
            }
        ]
    }
```
We need `Object.entries()` because deck.gl expects an array format:
![image](https://github.com/user-attachments/assets/9fd9dd61-32ea-4c69-9733-bebc7b5fb3aa)

Object.entries() converts the above object into an array of [key, value] pairs:
    
    [
        [
            "39.78-86.15",                              // d[0]: coordinate string
            [{id: 1, ...}, {id: 2, ...}]                // d[1]: array of incidents
        ],
        [
            "39.80-86.13",
            [{id: 3, ...}]
        ]
    ]